core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala

```diff
@@ -166,17 +166,6 @@ private[spark] class CoarseGrainedExecutorBackend(
       if (executor == null) {
         exitExecutor(1, "Received LaunchTask command but executor was null")
       } else {
-        if (decommissioned) {
```

Comment:
Are you sure that this case won't happen ? Or perhaps you mean that even if it does happen, it only lasts for a short duration: (until the Driver gets to know that the executor is decommissioned), that its not worth handling it ? If so, I agree with you. 

But just want to confirm when can this case even happen: task launched on a decommissioned executor ?

Reply:
It does happen but it's meaningless to send a decommission notification to driver at this point. Because we assume that we've already sent the notification when `decommissioned=true`(if decommission is triggered at executor).

It happens when `CoarseGrainedSchedulerBackend` handles `ReviveOffers` before `ExecutorDecommissioned`. (executor decommission can happen at any time so we can not guarantee the orders between `ReviveOffers` and `ExecutorDecommissioned`.)